lib/Input.js

```diff
@@ -37,7 +47,9 @@ class Input extends React.Component {
   }
 
   componentDidUpdate (prevProps) {
-    if (this.props.autoresize && prevProps.query !== this.props.query) {
+    if (prevProps.autoresize !== this.props.autoresize ||
+      (this.props.autoresize && RESIZE_PROPS.some((prop) => this.props[prop] !== prevProps[prop]))
+    ) {
```

Comment:
I'm not sure I totally follow. It sounds like that's effectively what the `RESIZE_PROPS` const does -- it's only updating the input width when a prop that may effect the width calculation is updated. I'm not sure effect checking if the calculated `inputWidth` (on state)  in `componentDidUpdate` would have. Seems it could prevent the `inputWidth` from being updated in cases where it needs to be recalculated?